includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp-subscription-list-trait.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * Service Provider: Mailchimp Subscription List Trait
+ *
+ * @package Newspack
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Extend the Subscription_List object with Mailchimp specific functionality.
+ *
+ * Since Mailchimp also offers Tags and Groups as Subscription Lists, some information is stored in the Public ID.
+ *
+ * This trait provides methods to extract the list type, the list ID and the Audience ID from the Public ID.
+ *
+ * All methods should be prefixed with `mailchimp_`.
+ */
+trait Newspack_Newsletters_Mailchimp_Subscription_List_Trait {
+
+	/**
+	 * Creates a list Public ID based on the type, the ID and the list (Audience) ID
+	 *
+	 * In Mailchimp, we offer both Audiences, Groups, and Tags as Subscription Lists. We modify the group and tag IDs so we can differentiate them from the Audiences IDs.
+	 *
+	 * Also, when working with groups or tags, we need to know the list ID, so we add it to the ID.
+	 *
+	 * @param string $item_id The item ID.
+	 * @param string $list_id The List/Audience ID.
+	 * @param string $type 'group' or 'tag'.
+	 * @return string
+	 */
+	public static function mailchimp_create_public_id( $item_id, $list_id, $type = 'group' ) {
```

Comment:
```suggestion
	public static function mailchimp_generate_public_id( $item_id, $list_id, $type = 'group' ) {
```

NIT suggestion, `create` feels like something is being added to a database.

Reply:
https://github.com/Automattic/newspack-newsletters/pull/1620/commits/bcfa2d2cbf7295480c0c4c3066951a79e7f94a6d